src/qibocal/auto/execute.py

```diff
@@ -133,24 +136,45 @@ def run(self, mode):
         self.head = self.graph.start
         while self.head is not None:
             task = self.current
-            log.info(f"Executing mode {mode.name} on {task.id}.")
+            task.iteration = self.history.iterations(task.id)
+            if self.history.iterations(task.id) > MAX_ITERATIONS:
+                raise_error(
+                    ValueError,
+                    f"Maximum number of iterations {MAX_ITERATIONS} reached!",
+                )
+            log.info(
+                f"Executing mode {mode.name} on {task.id} iteration {task.iteration}."
+            )
             completed = task.run(
                 platform=self.platform,
                 qubits=self.qubits,
                 folder=self.output,
                 mode=mode,
             )
             self.history.push(completed)
-            if isinstance(completed.status, Failure) and mode.name == "autocalibration":
-                log.warning("Stopping execution due to error in validation.")
-                yield task.uid
-                break
             self.head = self.next()
+
+            if isinstance(completed.status, Failure) and mode.name == "autocalibration":
+                if task.action.handler is None:
+                    log.warning("Stopping execution due to error in validation.")
+                    yield completed.task.uid
+                    break
+                else:
+                    log.info(
+                        f"Handling validation error by executing node {task.action.handler.id}"
+                    )
+
+                    self.head = task.action.handler.id
+                    yield completed.task.uid
+
             update = self.update and task.update
             if (
                 mode.name in ["autocalibration", "fit"]
                 and self.platform is not None
                 and update
             ):
-                task.update_platform(results=completed.results, platform=self.platform)
-            yield task.uid
+                completed.task.update_platform(
+                    results=completed.results, platform=self.platform
+                )
```

Comment:
Should we lift even the `update_platform`? All in all, it makes sense that a platform is updated by a completed task.
```suggestion
                completed.update_platform(self.platform)
```